ilastik/applets/dataSelection/opDataSelection.py

```diff
@@ -566,9 +572,16 @@ def from_h5_group(cls, group: h5py.Group, params: Dict = None):
 
     @staticmethod
     def _nickname_from_url(url: str) -> str:
+        """
+        Take the part after the last /, make it safe for use as a file name,
+        remove anything that looks like an extension ('.zarr') and replace remaining dots
+        to ensure exporting logic does not mistake them for file extensions.
+        """
         last_url_component = url.rstrip("/").rpartition("/")[2]
         filename_safe = re.sub(r"[^a-zA-Z0-9_.-]", "_", last_url_component)
-        return filename_safe
+        extensionless = os.path.splitext(filename_safe)[0]
+        exporter_safe = extensionless.replace(".", "_")
```

Comment:
I don't yet understand this one - Why is this only an issue with URL-type datasets? I tried with an image with lots of dots in the name and found no odd behavior on export.

I'm only asking because for users it might not be obvious behavior - for some datatypes the filename stem is preserved, for others not. Could you give an example filename that wouldn't work?

Reply:
![image](https://github.com/ilastik/ilastik/assets/63287233/919030d2-5207-4b2d-9677-bceb1f9051bd)

Here's an example: The file was called `photo.jpg.jpg`. The nickname became `photo.jpg`, and the default export location trims that second "extension" including the `_Probabilities` part that is supposed to be in the default export file name. So in fact not fixing this would have been consistent with existing behaviour...

Reply:
So should I revert or also fix it for the other `DatasetInfo` subclasses? 🙃

Reply:
In general though, we remove one file extension to create the nickname, so removing `.zarr` is canon, replacing `.` with `_` is a new feature.